src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -71,113 +71,52 @@ def _build_resource_id(**kwargs):
     except KeyError:
         return None
 
+def _rfc6901_decode(encoded):
+    return unquote(encoded).replace("~1", "/").replace("~0", "~")
+
+def _get_json_pointer_segments(json_pointer):
+    return [_rfc6901_decode(segment) for segment in json_pointer.split('/')]
+
+def _resolve_parameter_type(parameter, definitions):
```

Comment:
This logic is a translation of https://github.com/Azure/azure-powershell/blob/e0ceef904759fd32026e2de9c0693a1b9f264161/src/Resources/ResourceManager/Utilities/TemplateUtility.cs#L296